mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -1049,7 +1049,7 @@ object ALS extends DefaultParamsReadable[ALS] with Logging {
       .join(userFactors)
       .mapPartitions({ items =>
         items.flatMap { case (_, (ids, factors)) =>
-          ids.view.zip(factors)
+          ids.iterator.zip(factors.iterator)
```

Comment:
What's the warning BTW?  and does `ids.zip(factors)` not work here?

Reply:
I guess `.view` was used to avoid creating a new Array

Reply:
Hm, but it results in a new array already, so just removing .view seems to not change much here. Just in case @mengxr do you recall why? this was written 4 years ago, so may have been lost in time.

The result would now be an iterator, which seems OK; avoids allocating an array after the change, I believe.